File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& nums) {
        int rplc = nums[nums.size()-1];
        nums[nums.size()-1] = -1;
        for(int i=nums.size()-2;i>=0;i--)
        {
            int ele = nums[i];
            nums[i] = rplc;
            rplc = max(rplc,ele);
        }
        return nums;
    }
};

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
     
        int rplc = -1;
        int check = -1;
        
        for(int i=arr.length-1; i>=0;i--){
            if(check == -1){
                rplc = arr[i];
                arr[i] = -1;
                check = 1;
            }
            else{
                int temp = arr[i];
                arr[i] = rplc;
                if(rplc < temp){
                    rplc = temp;
                }
            }
        }
        
        return arr;
        
    }
}